packages/catalog-process/src/services/catalogService.ts

```diff
@@ -608,13 +603,11 @@ export async function deleteDocumentLogic({
   assertEServiceExist(eServiceId, eService);
   assertRequesterAllowed(eService.data.producerId, authData.organizationId);
 
-  const descriptor = eService.data.descriptors.find(
-    (d: Descriptor) => d.id === descriptorId
-  );
+  const descriptor = retrieveDescriptor(descriptorId, eService);
 
-  const document = (
-    descriptor ? [...descriptor.docs, descriptor.interface] : []
```

Comment:
It's no longer needed to check if descriptor is undefined, because now `eServiceDescriptorNotFound` is thrown at the lines above.